### PR TITLE
Build llvm in release mode; remove llvm-project before applying

### PR DESCRIPTION
**Pull request by 1ntEgr8**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35 [do not merge] Trigger docker image build on pr for testing
* **#34 Build llvm in release mode; remove llvm-project before applying**

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Review by samestep — APPROVED** [review]
Question: why doesn't the Push CI Docker Base / cilint-clang-tidy job get triggered by this PR? It seems like it should require a separate [do not merge] test PR in order to validate

File: ci-docker-base/Dockerfile.cilint-clang-tidy
Comment (janeyx99) [inline ci-docker-base/Dockerfile.cilint-clang-tidy:15]: small nit for clarity
```suggestion
# Build clang-tidy, copy out the binary, and remove the llvm checkout
```

**Review by janeyx99 — APPROVED** [review]
looks good :)

**Comment by janeyx99** [comment]
> Question: why doesn't the Push CI Docker Base / cilint-clang-tidy job get triggered by this PR? It seems like it should require a separate [do not merge] test PR in order to validate

I think the reason is because it pushes after it builds. In the long term, it would be a good idea to separate the push and build step so that PRs could test the build, but the actual push would happen after the PR is merged. An example could be found in the builder repo: https://github.com/pytorch/builder/blob/master/.github/workflows/build-libtorch-images.yml

**Comment by samestep** [comment]
> In the long term, it would be a good idea to separate the push and build step so that PRs could test the build, but the actual push would happen after the PR is merged.

+1 to this; another way to do it (which doesn't require a separate `env` variable) is to just add this line to the push step:
```yaml
if: github.ref == 'refs/heads/master'
```
Examples:
- https://github.com/pytorch/pytorch-ci-hud/blob/b596dfe378b3881f58b66d937e93f811497830e1/.github/workflows/ci.yml#L20
- https://github.com/pytorch/pytorch-probot/blob/a5c9035af666e816ed6b6a1d81cbffde9ceb0688/.github/workflows/build.yml#L18

**Pull request merged**
